vignettes/epidist.Rmd

```diff
@@ -1,201 +1,250 @@
+
+* `case`: 
+* `ptime`:
+* `stime`:
+
+Here we simulate data in this format, and in doing so explain the two main issues with observational delay data.
+<!-- Please see as of yet unwritten vignette for information about how to transform your data of other formats to the right format. -->
+
+First, we use the [Gillepsie algorithm](https://en.wikipedia.org/wiki/Gillespie_algorithm) to generate infectious disease outbreak data (Figure \@ref(fig:outbreak)) from a stochastic compartmental model:
+
+```{r}
 outbreak <- simulate_gillespie(seed = 101)
 ```
 
-Define the secondary distribution to use in the simulation
+(ref:outbreak) Early on in the epidemic, there is a high rate of growth in new cases. As more people are infected, the rate of growth slows. (Only every 50th case is shown to avoid over-plotting.)
 
-```{r secondary-dist}
-secondary_dist <- data.table(
-  meanlog = 1.8, sdlog = 0.5
-) |>
+```{r outbreak, fig.cap="(ref:outbreak)"}
+outbreak[case %% 50 == 0, ] |>
+  ggplot(aes(x = ptime, y = case)) +
+  geom_point(col = "#56B4E9") +
+  labs(x = "Primary event time (day)", y = "Case number") +
+  theme_minimal()
+```
+
+`outbreak` is a [`data.table`](https://cran.r-project.org/web/packages/data.table/index.html) with the columns `case` and `ptime`.
+
+Now, to generate secondary events, we will use a lognormal distribution (Figure \@ref(fig:lognormal)) for the delay between primary and secondary events:
+
+```{r}
+secondary_dist <- data.table(meanlog = 1.8, sdlog = 0.5) |>
   add_natural_scale_mean_sd()
 ```
 
-Simulate an observation process during the growth phase for a secondary event using a lognormal distribution, and finally simulate observing this event.
+(ref:lognormal) The lognormal distribution is skewed to the right. Long delay times still have some probability.
 
-```{r simulate-data}
+```{r lognormal, fig.cap="(ref:lognormal)"}
+ggplot(data.frame(x = c(0, 8)), aes(x = x)) +
+  geom_function(
+    fun = dlnorm,
+    meanlog = secondary_dist[["meanlog"]],
+    sdlog = secondary_dist[["sdlog"]]
+  ) +
+  theme_minimal() +
+  labs(
+    x = "Delay between primary and secondary event (days)",
+    y = "Probability density"
+  )
+```
+
+```{r}
 obs <- outbreak |>
   simulate_secondary(
-    meanlog = secondary_dist$meanlog[[1]],
-    sdlog = secondary_dist$sdlog[[1]]
-  ) |>
-  observe_process()
+    meanlog = secondary_dist[["meanlog"]],
+    sdlog = secondary_dist[["sdlog"]]
+  )
+```
+
+(ref:delay) Secondary events (in green) occur with a delay drawn from the lognormal distribution (Figure \@ref(fig:lognormal)). As with Figure \@ref(fig:outbreak), to make this figure easier to read, only every 50th case is shown.
+
+```{r delay, fig.cap="(ref:delay)"}
+obs[case %% 50 == 0, ] |>
+  ggplot(aes(y = case)) +
+  geom_segment(
+    aes(x = ptime, xend = stime, y = case, yend = case),
+    col = "grey"
+  ) +
+  geom_point(aes(x = ptime), col = "#56B4E9") +
```

Comment:
I would usually do this by going to long form data and then using the new variable as then you can have a legend. This is nice and clear as is though

Reply:
I would also usually do that but in some sense I was trying to make the number of lines used low.